include/onnxruntime/core/framework/ort_value.h

```diff
@@ -75,6 +75,22 @@ struct OrtValue {
 #endif
   }
 
+  bool IsInvalid() const noexcept {
+    return (type_ != nullptr && type_->IsInvalidType());
+  }
+
+  bool IsNonTensor() const noexcept {
+    return (type_ != nullptr && type_->IsNonTensorType());
+  }
+
+  bool IsOptional() const noexcept {
+    return (type_ != nullptr && type_->IsOptionalType());
+  }
```

Comment:
Optional data type can only exist in metadata. It can not be created at runtime, and therefore OrtValue can not contain it.